### PR TITLE
[4.0][RFC] hide db credentials from API

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #29196 .

### Summary of Changes
hide db credential fields

### Testing Instructions
`curl -H "Authorization: Bearer YOUR_TOKEN" {{URL}}/api/index.php/v1/config/application
`

### Expected result
db credentials not exposed

### Actual result
exposed

### Documentation Changes Required
?

### CODE REVIEW

File: api/components/com_config/src/View/Application/JsonapiView.php [hunk 1]
```diff
@@ -25,6 +25,29 @@
  */
 class JsonapiView extends BaseApiView
 {
+	/**
+	 * The fields to not render for secuty reason
```
Comment (Quy) [inline api/components/com_config/src/View/Application/JsonapiView.php]:
```suggestion
	 * The fields to not render for security reason
```

File: api/components/com_config/src/View/Application/JsonapiView.php [hunk 2]
```diff
@@ -25,6 +25,29 @@
...
 {
+	/**
+	 * The fields to not render for security reason
+	 *
+	 * @var  array
+	 * @since  4.0.0
+	 */
+	protected $fieldsToNotRender = [
+		'editor',
+		'captcha',
```
Comment (Quy) [inline api/components/com_config/src/View/Application/JsonapiView.php]: Why exclude?

Comment (alikon) [reply api/components/com_config/src/View/Application/JsonapiView.php]: exposing captcha info may help hacker to fine tune their attempt ?

the list is open .... 
it's a rfc after all :smile:

**Comment by brianteeman** [comment]
No need to reinvent the wheel. The "safe" data has already been defined for use by the admin sysinfo

https://github.com/joomla/joomla-cms/blob/8cdb9bd01fd5fbc37c3bb17b21e45d9eeb80218a/administrator/components/com_admin/models/sysinfo.php#L163

**Comment by alikon** [comment]
good catch

File: api/components/com_config/src/View/Application/JsonapiView.php [hunk 3]
```diff
@@ -25,6 +26,7 @@
  */
 class JsonapiView extends BaseApiView
 {
+
```
Comment (Quy) [inline api/components/com_config/src/View/Application/JsonapiView.php]:
```suggestion
```

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on d51fb2293dae9f095b49284ffc67ff35e0eee996<br /><br />Tested in Nightly Build of May 23 successfully.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29198">issues.joomla.org/tracker/joomla-cms/29198</a>.</sub>

**Review by zero-24 — APPROVED** [review]
LGTM thanks @alikon

**Comment by wilsonge** [comment]
Made this a draft so it's clear to others not to merge this until discussions in original issue are complete

**Comment by zero-24** [comment]
Well I personally would say that this PR here that is just hiding the sensible fields for now is the right thing todo. Removing the endpoint at all is discussed in the other issue. Or do you think we should expose the sensible fields via the API too? We even don't do that in the UI for all fields.

**Comment by wilsonge** [comment]
Yes. I'd expose the fields if you have the permissions to view the fields. You have to be a super admin. so why not? Database in the API allows you to do proper DevOps deploys of your website from say a staging to a production site (restore a backup and update the global configuration to the Prod values)

**Comment by particthistle** [comment]
Reading https://github.com/joomla/joomla-cms/issues/29196 which was closed June 3 should this now be closed @alikon to clear it out of the patch tester and pending PR list? 

_Still getting the hang of the protocol on things here so feel free to correct my perspective if that's not how these types of things are managed._ 

Discovered during PBF July 18 while testing other API PRs.

**Comment by richard67** [comment]
I am a bit confused now. Issue #29196 has been closed as intended behavior. But maybe this PR still makes sense, now not as a bug fix but as a new feature? The discussion above seems to be open to me, but also stale, nothing new since a while. @alikon @wilsonge @zero-24 how to continue here?

**Comment by alikon** [comment]
i really don't know what i can do more 
i've opened this pr as a RFC cause of #29196
after, it has been demoted to draft 
after #29196 has been closed
...after 2 months...
so i guess the best think i can do 
it is simply to close this one

**Pull request closed without merging**